File: java/Calculator.3.java
import java.util.*;
import java.io.File;
import java.io.FileReader;
import java.io.BufferedReader;

// Calculator  class
public class Calculator {
	// History of results
	private List<Float> history = new ArrayList<Float>();
	
	// Constructor for loading results from file
	public Calculator(String fileName) {
		this.loadHistoryFromFile(fileName);
		this.print("Calculator was created");
	}
	
	// Returns if number is odd
	public boolean isOdd(float number) {
		boolean boolIsOdd = (number%2) != 0;
		switch (boolIsOdd) {
			case true:
				this.print(number + " is odd!");
				return true;
			case false: 
				this.print(number + " is not odd!");
				return false;
		}
	}
	
	// Adds two numbers
	public float addition(float number1, float number2) {
		float result = number1 + number2;
		this.print("Result od addition is " + result);
		this.addToHistory(result);
		return result;
	}
	
	// Compare two numbers
	public void compareNumbers(float number1, float number2) {
		if (number1 > number2) {
			this.print(number1 + " is bigger than " + number2);
		} else if (number2 > number1) {
			this.print(number1 + " is smaller than " + number2);
		} else {
			this.print(number1 + " is the same as " + number2);
		}
	}
	
	// Subtracts two numbers
	public float substraction(float number1, float number2) {
		float result = number1 - number2;
		this.print("Result od substraction is " + result);
		this.addToHistory(result);
		return result;
	}
	
	// Print the whole history
	public void printHistory() {
		for (float result: this.history) {
			this.print(String.valueOf(result));
		}
	}
	
	// Multiplies two numbers
	public float multiplication(float number1, float number2) {
		float result = number1 * number2;
		this.print("Result od multiplication is " + result);
		this.addToHistory(result);
		return result;
	}
	
	// Adds result to history array
	private void addToHistory(float item) {
		this.history.add(item);
	}
	
	// Divides two numbers
	public float division(float number1, float number2) {
		float result = number1 / number2;
		this.print("Result od division is " + result);
		this.addToHistory(result);
		return result;
	}
	
	// Prints text with current date
	public void print(String text) {
		java.util.Date date = new java.util.Date();
	    System.out.println("# " + date.toString() + " # " + text );
	}

	// Adds results to history
	public void loadHistoryFromFile(String fileName) {
		try {
			File historyFile = new File(fileName);
			BufferedReader reader = new BufferedReader(new FileReader(historyFile));
            String readLine = "";
            while ((readLine = reader.readLine()) != null) {
                this.addToHistory(Float.parseFloat(readLine));
            }
            reader.close();
            this.print("History form file was loaded!");
		} catch (Exception e) {
			this.print("File could not be loaded! Because:" + e);
		}
	}
}

File: python/Calculator.3.py
import datetime

class Calculator:
    # History of results
    history = []

    # Constructor for loading results from file
    def __init__(self, fileName):
        self.loadHistoryFromFile(fileName)
        self.print("Calculator was created")

    # Returns if number is odd
    def isOdd(self, number):
        boolIsOdd = number / 2 != 0
        if boolIsOdd:
            self.print(str(number) + " is odd!")
            return True
        if not boolIsOdd:
            self.print(str(number) + " is not odd!")
            return False

    # Add two numbers
    def addition(self, number1, number2):
        result = number1 + number2
        self.print('Result of addition is ' + str(result))
        self.addToHistory(result)
        return result

    # Print the whole history
    def printHistory(self):
        for result in self.history:
            self.print(str(result))

    # Substracts two numbers
    def substraction(self, number1, number2):
        result = number1 - number2
        self.print('Result of substraction is ' + str(result))
        self.addToHistory(result)
        return result

    # Compare two numbers
    def compareNumbers(self, number1,  number2):
        if number1 > number2:
            self.print(str(number1) + " is bigger than " + str(number2))
        elif number2 > number1:
            self.print(str(number1) + " is smaller than " + str(number2))
        else:
            self.print(str(number1) + " is the same as " + str(number2))

    # Multiplies two numbers
    def multiplication(self, number1, number2):
        result = number1 * number2
        self.print('Result of multiplication is ' + str(result))
        self.addToHistory(result)
        return result

    # Adds result to history array
    def addToHistory(self, item):
        self.history.append(item)
 
    # Divides two numbers
    def division(self, number1, number2):
        result = number1 / number2
        self.print('Result of division is ' + str(result))
        self.addToHistory(result)
        return result

    # Prints text with current date
    def print(self, text):
        date = datetime.datetime.now()
        print("# " + str(date) + " # " + text)

    # Adds  results  to history
    def loadHistoryFromFile(self, fileName):
        try:
            file = open(fileName)
            readLine = file.readline()
            while readLine != None:
                self.addToHistory(readLine)
                readLine = file.readline()
            file.close()
            self.print("History from file was loaded!")
        except Exception:
            self.print("File could not be loaded! Because:" + str(Exception))

